lightning/src/ln/channelmanager.rs

```diff
@@ -10953,7 +10955,12 @@ where
 			},
 			OffersMessage::InvoiceError(invoice_error) => {
 				let payment_hash = match context {
-					Some(OffersContext::InboundPayment { payment_hash }) => Some(payment_hash),
+					Some(OffersContext::InboundPayment { payment_hash, nonce, hmac }) => {
+						match signer::verify_payment_hash(payment_hash, hmac, nonce, expanded_key) {
+							Ok(_) => Some(payment_hash),
+							Err(_) => None,
+						}
```

Comment:
Could you update these to use `pub` methods on `PaymentHash` as per #3242?

Reply:
Thanks a lot for the suggestion!

I’ve been experimenting with implementing `pub` methods on `PaymentHash`, drawing some inspiration from #3242. However, it looks like this might be a bit more complicated than I initially thought.

The challenge arises because `PaymentHash` is part of the `lightning-types` crate, which is a dependency of the `lightning` crate. Plus, the `hmac_for_offer_payment` and `verify` methods need `ExpandedKey`, `HMAC`, and the `signer` struct, which brings us into a circular dependency situation.

So, we have a couple of options:

- We could move `PaymentHash` from the lightning-types crate to the lightning crate, or
- Implement a wrapper struct for `PaymentHash`.

I’d love to hear your thoughts on which direction you think we should take!

Thank you very much!

Reply:
Another option is to include a trait in `lightning` for these methods and implement it on `PaymentHash`.

Reply:
That's sounds like a much cleaner idea!
Introduce the `Verification` trait in [pr3163.11](https://github.com/shaavan/rust-lightning/commits/pr3163.11)

Thank you so much for the suggestion! 🌟